src/blog/2024/05/exploring-node-red-dashboard-2-widgets.md

```diff
@@ -0,0 +1,220 @@
+```
+!["Screenshot displaying the change node setting JSON expression to payload for retrieving and sorting data."](./images/exploring-dashboard-2-widgets-change-node.png "Screenshot displaying the change node setting JSON expression to payload for retrieving and sorting data."){data-zoomable}
+
+### Displaying Data on the Table
+
+To display data on the table, we use the **ui-table** widget in Dashboard 2.0. This widget accepts an array of objects as input. The columns in the table correspond to the properties of the objects within the array, and each row represents a different object with values corresponding to those properties.
+
+1. Drag a **ui-table** widget onto the canvas.
+2. Create a new **ui-page** and **ui-group** for it.
+3. Connect the output of the **change** node to the input of the **ui-table** widget.
+
+!["Screenshot displaying the ui-table widget configuration"](./images/exploring-dashboard-2-widgets-table-widget.png "Screenshot displaying the ui-table widget configuration"){data-zoomable}
+
+For more information on ui-table refer to [ui-table docs](https://dashboard.flowfuse.com/nodes/widgets/ui-table.html)
+
+### Retrieve, Calculate and format income-expense data
+
+In our application, we will display data on a chart, showing the total income and total expenses for analysis. This section will focus on retrieving, formatting, and calculating the necessary data for charting.
+
+1. Drag the two **change** node onto the canvas.
+2. For the first **Change** node Set `msg.payload` to `global.income` and `msg.topic` to "income" and give it name "retrive income". For the second **Change** node, set `msg.payload` to `global.expense` and `msg.topic` to "expense" and give that second change node name "retrive expense".
+
+!["Screenshot displaying the change node retrieving income data from global context"](./images/exploring-dashboard-2-widgets-chart-widget-retrieve-income-change-node.png "Screenshot displaying the change node retrieving income data from global context"){data-zoomable}
+
+!["Screenshot displaying the change node retrieving expense data from global context"](./images/exploring-dashboard-2-widgets-chart-widget-retrieve-expense-change-node.png "Screenshot displaying the change node retrieving expense data from global context"){data-zoomable}
+
+3. Drag a **Split** node onto the canvas.
+4. Drag the **Change** node onto the canvas and set `msg.payload.amount` to the JSON expression `$number(payload.amount)` and give it name "Convert amount to number".
+
+!["Screenshot displaying the change node converting amount to number"](./images/exploring-dashboard-2-widgets-chart-widget-convert-amount-to-number-change-node.png "Screenshot displaying the change node converting amount to number"){data-zoomable}
+
+5. Drag a **Join** node onto the canvas select mode to reduced expression and set the **Reduce exp** to `$A + payload.amount`, **Initial value** to 0, and **Fix-up exp** to `$A` and give this join node name "Calculate total".
```

Comment:
not clear to me here what `$A` is?

Reply:
It's a special variable often used in the join node to store accumulated values. Other variables used in the join node include:

`$I`: index of the message in the sequence,
`$N`: number of messages in the sequence.

Reply:
Can you mention that here please? Even as someone using NR for years, I never know about that

Reply:
Yes, I will mention it in the blog. I was also unaware of this until I started working on the core node documentation.